src/java.desktop/windows/classes/sun/print/PrintServiceLookupProvider.java

```diff
@@ -116,6 +116,7 @@ private synchronized void refreshServices() {
         if (printers == null) {
             // In Windows it is safe to assume no default if printers == null so we
             // don't get the default.
+            invalidateServices();
```

Comment:
The default printer service, `defaultPrintService`, is usually in `printServices` array. It's possible that it's not in the array; in this case getDefaultPrintService() will reset it to `null` when accessed.